### PR TITLE
[Wasm] Forced filesystem creation

### PR DESCRIPTION
**Pull request by MarcosCobena**
I'm looking for forcing Emscripten to enable the filesystem even if emcc didn't detect such during compilation. This enables scenarios where users can preload files through Emscripten's file_packager.py tool, without needing to rebuild mono.wasm & co. by them-selves.

### CODE REVIEW

**Comment by MarcosCobena** [comment]
Ideally, those are the steps needed to repro:

https://github.com/MarcosCobena/WasmApp1/blob/file-packager-sample/WasmApp1/Program.cs#L10

; however, although I can see the file to be mounted by evaluating "FS" through browser's Console (for this reason I've in advance created the PR), I haven't run successfully above end to end: mono.wasm and mono.js from my own build reports an unsync between the DLLs at bin/Debug/managed and, I guess, the ones I've built my-self.

Summing up: I can add above test to sdks/wasm root, which's maybe a better place...

**Comment by MarcosCobena** [comment]
@kg thanks for your comment! The PR which introduced --preload-file already turned on that flag implicitly. Also, I've been reading Emscripten docs and haven't found any downside, although I guess there's a little bit of overhead when initializing mono runtime...

**Comment by kjpou1** [comment]
I also have a pr to turn this on for TimeZoneinformation. This looks good to me.

**Comment by kjpou1** [comment]
Other PR https://github.com/mono/mono/pull/17760. This also shows that the runtime does not increase in size.

File: sdks/wasm/Makefile [hunk 1]
```diff
@@ -126,7 +126,7 @@ MONO_LIBS = $(TOP)/sdks/out/wasm-runtime-release/lib/{libmono-ee-interp.a,libmon
 MONO_THREADS_LIBS = $(TOP)/sdks/out/wasm-runtime-threads-release/lib/{libmono-ee-interp.a,libmono-native.a,libmono-icall-table.a,libmonosgen-2.0.a,libmono-ilgen.a}
 MONO_DYNAMIC_LIBS = $(TOP)/sdks/out/wasm-runtime-dynamic-release/lib/{libmono-ee-interp.a,libmono-native.a,libmono-icall-table.a,libmonosgen-2.0.a,libmono-ilgen.a}
 
-EMCC_FLAGS=-s WASM=1 -s ALLOW_MEMORY_GROWTH=1 -s BINARYEN=1 -s ALIASING_FUNCTION_POINTERS=0 -s NO_EXIT_RUNTIME=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'FS_createPath', 'FS_createDataFile', 'cwrap', 'setValue', 'getValue', 'UTF8ToString', 'addFunction']" -s USE_ZLIB=1 -s "EXPORTED_FUNCTIONS=['_putchar']" --source-map-base http://example.com  -s WASM_OBJECT_FILES=0
+EMCC_FLAGS=-s WASM=1 -s ALLOW_MEMORY_GROWTH=1 -s BINARYEN=1 -s ALIASING_FUNCTION_POINTERS=0 -s NO_EXIT_RUNTIME=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'FS_createPath', 'FS_createDataFile', 'cwrap', 'setValue', 'getValue', 'UTF8ToString', 'addFunction']" -s USE_ZLIB=1 -s "EXPORTED_FUNCTIONS=['_putchar']" --source-map-base http://example.com  -s WASM_OBJECT_FILES=0 -s FORCE_FILESYSTEM=1
```
Comment (MarcosCobena) [inline sdks/wasm/Makefile]: @kjpou1 shouldn't this then be reverted? I added this here to force every mono.wasm ever built to include fs support, but we now have --enable-fs at packager.exe level. I'd like to know your feelings here :-)

Another important reason is we consume mono.wasm through your NuGets from Jenkins (nowadays), and want to preload stuff in a target or similar by calling file_packager.py, so we need such mono.wasm to have fs support.

Comment (kjpou1) [reply sdks/wasm/Makefile]: No, the runtime built from the makefile will be packaged through the NuGets and the SDK as well so it needs to be set here so that separate packaged files can be loaded.

The packager.exe is a separate process right now that builds its own runtime modules to be used.  If anything changes in the future we will have that option there to control export the filesystem methods.  When one uses Emscripten with the --preload-file option it will automatically add the FORCE_FILESYSTEM filesystem for you.

Comment (MarcosCobena) [reply sdks/wasm/Makefile]: Cristal clear, thanks man :-)

**Comment by MarcosCobena** [comment]
@kjpou1 you may like this: https://gist.github.com/MarcosCobena/0e2ebd948fd51c868693f9655f527928

We're using it for packaging the entire app content in Wave Engine :-)
```
  <Target Name="FilePackagingScript" AfterTargets="AfterBuild">
    <PropertyGroup>
      <BuildCommand>python ../../../Scripts/file_packager.py $(OutDir)/mono.dat --js-output="$(OutDir)/mono-loader.js" --preload "$(OutDir)/Content@/Content"</BuildCommand>
    </PropertyGroup>
    <Exec Command="$(BuildCommand)" />
    <RemoveDir Directories="$(OutDir)/Content" />
  </Target>
```

**Comment by kjpou1** [comment]
@MarcosCobena actually very interested in seeing how it performs with the pre-load files.

I was going to ping you guys once the my PR went in to see if you could use something like that as well.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build

**Comment by MarcosCobena** [comment]
I've tested the latest NuGets from Jenkins and it works! However, we need a small fix just because our Content/ is larger than 16 MB:

https://github.com/mono/mono/pull/17829
